devops/containers/ubuntu2404_build_oneapi.Dockerfile

```diff
@@ -1,48 +1,21 @@
-FROM nvidia/cuda:12.6.3-devel-ubuntu24.04
+FROM ghcr.io/intel/llvm/ubuntu2404_build
 
 ENV DEBIAN_FRONTEND=noninteractive
 
 USER root
 
-# Install SYCL prerequisites
-COPY scripts/install_build_tools.sh /install.sh
-RUN /install.sh
-
-SHELL ["/bin/bash", "-ec"]
-
 # Install oneAPI
 
-# Make the directory if it doesn't exist yet.
-# This location is recommended by the distribution maintainers.
-RUN mkdir --parents --mode=0755 /etc/apt/keyrings
-# Download the key, convert the signing-key to a full
-# keyring required by apt and store in the keyring directory
-RUN wget https://repo.radeon.com/rocm/rocm.gpg.key -O - | \
-gpg --dearmor | tee /etc/apt/keyrings/rocm.gpg > /dev/null && \
-# Add rocm repo
-echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/amdgpu/6.3/ubuntu noble main" \
-    | tee /etc/apt/sources.list.d/amdgpu.list && \
-echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/6.3 noble main" \
-    |  tee --append /etc/apt/sources.list.d/rocm.list && \
-echo -e 'Package: *\nPin: release o=repo.radeon.com\nPin-Priority: 600' \
-    | tee /etc/apt/preferences.d/rocm-pin-600 && \
-echo -e 'Package: *\nPin: release o=repo.radeon.com\nPin-Priority: 600' \
-    | tee /etc/apt/preferences.d/rocm-pin-600 && \
-wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor \
+RUN wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor \
 | tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null && \
 echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" \
 | tee /etc/apt/sources.list.d/oneAPI.list
 
 # Install the ROCM kernel driver and oneAPI
-RUN apt update && apt install -yqq rocm-dev intel-oneapi-compiler-dpcpp-cpp && \
+RUN apt update && apt install -yqq intel-oneapi-compiler-dpcpp-cpp && \
```

Comment:
Installing DPC++ using `apt` will install the latest public release of DPC++, right?

Reply:
Correct, I could guard the version so it doesn't auto-update but I don't expect new releases to break anything. If they do, I'll add the guard.

Reply:
> If they do, I'll add the guard.

I think we should add the version guard proactively, perhaps in a separate PR. This would save us the effort of debugging unrelated failures if newer DPC++ releases break anything.

Reply:
Sure. it's easy enough, I can add it here.